src/spikeinterface/core/waveforms_extractor_backwards_compatibility.py

```diff
@@ -543,7 +544,11 @@ def _read_old_waveforms_extractor_binary(folder, sorting):
         ext = new_class(sorting_analyzer)
         with open(ext_folder / "params.json", "r") as f:
             params = json.load(f)
-        ext.params = params
+        # update params
+        new_params = ext._set_params()
+        updated_params = make_ext_params_up_to_date(ext, params, new_params)
```

Comment:
added this here because if we don't do this old waveform extractors loaded as analyzers won't be able to undergo merging because of conflicts with parameters.